In [1]:
import pandas as pd

In [2]:
klaeger = pd.read_excel('../data/KlaegerScience2017/Klaeger_Science_2017 Supplementary Table 2 Target Lists.xlsx', sheet_name='Kinobeads')

In [3]:
klaeger

Drug           Lysate      Beads Gene Name  \
0     Abemaciclib  4 cell line mix  Kinobeads      AAK1   
1     Abemaciclib  4 cell line mix  Kinobeads     ADCK1   
2     Abemaciclib  4 cell line mix  Kinobeads     BMP2K   
3     Abemaciclib  4 cell line mix  Kinobeads    CAMK2D   
4     Abemaciclib  4 cell line mix  Kinobeads    CAMK2G   
...           ...              ...        ...       ...   
5911      Y-39983  4 cell line mix  Kinobeads    Q6ZSR9   
5912      Y-39983  4 cell line mix  Kinobeads     ROCK1   
5913      Y-39983  4 cell line mix  Kinobeads     ROCK2   
5914      Y-39983  4 cell line mix  Kinobeads      SIK3   
5915      Y-39983  4 cell line mix  Kinobeads     TAOK1   

      Relative Intensity DMSO  Relative Intensity 3 nM  \
0                           1                 0.905556   
1                           1                 1.045974   
2                           1                 1.101696   
3                           1                 0.894551   
4                           1                 1.125435   
...                       ...                      ...   
5911                        1                 1.363107   
5912                        1                 0.862450   
5913                        1                 0.857286   
5914                        1                 0.705010   
5915                        1                 1.232738   

      Relative Intensity 10 nM  Relative Intensity 30 nM  \
0                     0.791054                  0.738292   
1                     0.869866                  1.141565   
2                     1.012297                  1.087891   
3                     0.984481                  0.900583   
4                     1.565245                  1.236451   
...                        ...                       ...   
5911                  1.255217                  1.096432   
5912                  0.436086                  0.163894   
5913                  0.489824                  0.272601   
5914                  0.449942                  0.534545   
5915                  1.329797                  1.364373   

      Relative Intensity 100 nM  Relative Intensity 300 nM  ...    Bottom  \
0                      0.639565                   0.323354  ...  0.009092   
1                      0.000000                   0.000000  ... -0.001200   
2                      0.973740                   0.863230  ... -1.003278   
3                      0.440384                   0.178266  ...  0.018690   
4                      0.315507                   0.139777  ...  0.043265   
...                         ...                        ...  ...       ...   
5911                   0.624299                   0.385381  ... -0.001292   
5912                   0.043354                   0.019142  ...  0.013883   
5913                   0.116512                   0.056129  ...  0.017236   
5914                   0.429215                   0.444114  ... -0.124816   
5915                   1.231199                   1.174358  ... -0.003968   

           Top    Inflection          EC50 EC50 Standard Error  \
0     0.957905    151.288760    151.288760           45.696999   
1     1.014355     63.935536     63.935536           68.832970   
2     1.034558  84774.108960  84774.108960       217756.243215   
3     0.972850     95.373379     95.373379           12.131433   
4     1.232332     82.363844     82.363844                 NaN   
...        ...           ...           ...                 ...   
5911  1.227442    139.520451    139.520451           64.178464   
5912  1.008762      8.649046      8.649046            0.414923   
5913  1.014790     10.646567     10.646567            1.206882   
5914  0.991565     86.205013     86.205013          170.714149   
5915  1.234836   3053.341142   3053.341142         2455.018786   

      Correction Factor   Apparent Kd        R2        BIC  \
0              0.673073    101.828368  0.985386 -20.503519   
1              0.622595     39.805937  0.983

In [4]:
5916*26

153816

In [62]:
kd = klaeger.melt(id_vars=['Drug', 'Gene Name'],
        value_vars=['Relative Intensity 10 nM',
                   'Relative Intensity 100 nM',
                   'Relative Intensity 3 nM',
                   'Relative Intensity 30 nM',
                   'Relative Intensity 300 nM',
                   'Relative Intensity 1000 nM',
                   'Relative Intensity 3000 nM',
                   'Relative Intensity 30000 nM'])
kd.columns = ['Drug','Gene','Concentration','Value']
kd = kd.pivot_table(index=['Drug','Concentration'], columns='Gene').fillna(1.0)
kd.head()

Value                                       \
Gene                               AAK1 ABL1 ABL2 ACAD10 ACAD11 ACADVL ACOX1   
Drug   Concentration                                                           
AC-480 Relative Intensity 10 nM     1.0  1.0  1.0    1.0    1.0    1.0   1.0   
       Relative Intensity 100 nM    1.0  1.0  1.0    1.0    1.0    1.0   1.0   
       Relative Intensity 1000 nM   1.0  1.0  1.0    1.0    1.0    1.0   1.0   
       Relative Intensity 3 nM      1.0  1.0  1.0    1.0    1.0    1.0   1.0   
       Relative Intensity 30 nM     1.0  1.0  1.0    1.0    1.0    1.0   1.0   

                                                    ...                    \
Gene                              ACOX3 ACP1 ACSL5  ... ULK3 UNC119 VDAC3   
Drug   Concentration                                ...                     
AC-480 Relative Intensity 10 nM     1.0  1.0   1.0  ...  1.0    1.0   1.0   
       Relative Intensity 100 nM    1.0  1.0   1.0  ...  1.0    1.0   1.0   
       Relative Intensity 1000 nM   1.0  1.0   1.0  ...  1.0    1.0   1.0   
       Relative Intensity 3 nM      1.0  1.0   1.0  ...  1.0    1.0   1.0   
       Relative Intensity 30 nM     1.0  1.0   1.0  ...  1.0    1.0   1.0   

                                                                            \
Gene                              WEE1 YARS YES1 YTHDF3;YTHDF1 YWHAG YWHAQ   
Drug   Concentration                                                         
AC-480 Relative Intensity 10 nM    1.0  1.0  1.0           1.0   1.0   1.0   
       Relative Intensity 100 nM   1.0  1.0  1.0           1.0   1.0   1.0   
       Relative Intensity 1000 nM  1.0  1.0  1.0           1.0   1.0   1.0   
       Relative Intensity 3 nM     1.0  1.0  1.0           1.0   1.0   1.0   
       Relative Intensity 30 nM    1.0  1.0  1.0           1.0   1.0   1.0   

                                        
Gene                               ZAK  
Drug   Concentration                    
AC-480 Relative Intensity 10 nM    1.0  
       Relative Intensity 100 nM   1.0  
       Relative Intensity 1000 nM  1.0  
       Relative Intensity 3 nM     1.0  
       Relative Intensity 30 nM    1.0  

[5 rows x 520 columns]

In [12]:
data = pd.melt(klaeger, id_vars=['Drug', 'Gene Name'],
        value_vars=['Relative Intensity 10 nM',
                   'Relative Intensity 100 nM',
                   'Relative Intensity 3 nM',
                   'Relative Intensity 30 nM',
                   'Relative Intensity 300 nM',
                   'Relative Intensity 1000 nM',
                   'Relative Intensity 3000 nM',
                   'Relative Intensity 30000 nM'])

In [10]:
klaeger.columns

Index(['Drug', 'Lysate', 'Beads', 'Gene Name', 'Relative Intensity DMSO',
       'Relative Intensity 3 nM', 'Relative Intensity 10 nM',
       'Relative Intensity 30 nM', 'Relative Intensity 100 nM',
       'Relative Intensity 300 nM', 'Relative Intensity 1000 nM',
       'Relative Intensity 3000 nM', 'Relative Intensity 30000 nM',
       'DMSO Intensity', 'Intensity Type', 'Slope', 'Bottom', 'Top',
       'Inflection', 'EC50', 'EC50 Standard Error', 'Correction Factor',
       'Apparent Kd', 'R2', 'BIC', 'Target Classification'],
      dtype='object')

In [13]:
data.columns = ['Drug','Gene','Concentration','Value']

In [14]:
data

Drug    Gene                Concentration     Value
0      Abemaciclib    AAK1     Relative Intensity 10 nM  0.791054
1      Abemaciclib   ADCK1     Relative Intensity 10 nM  0.869866
2      Abemaciclib   BMP2K     Relative Intensity 10 nM  1.012297
3      Abemaciclib  CAMK2D     Relative Intensity 10 nM  0.984481
4      Abemaciclib  CAMK2G     Relative Intensity 10 nM  1.565245
...            ...     ...                          ...       ...
47323      Y-39983  Q6ZSR9  Relative Intensity 30000 nM  0.000000
47324      Y-39983   ROCK1  Relative Intensity 30000 nM  0.014285
47325      Y-39983   ROCK2  Relative Intensity 30000 nM  0.006793
47326      Y-39983    SIK3  Relative Intensity 30000 nM  0.000000
47327      Y-39983   TAOK1  Relative Intensity 30000 nM  0.000000

[47328 rows x 4 columns]

In [17]:
data = data.pivot_table(index=['Drug','Concentration'], columns='Gene').fillna(1.0)

In [45]:
data.shape

(1824, 520)

In [61]:
data.head()

Value                                       \
Gene                               AAK1 ABL1 ABL2 ACAD10 ACAD11 ACADVL ACOX1   
Drug   Concentration                                                           
AC-480 Relative Intensity 10 nM     1.0  1.0  1.0    1.0    1.0    1.0   1.0   
       Relative Intensity 100 nM    1.0  1.0  1.0    1.0    1.0    1.0   1.0   
       Relative Intensity 1000 nM   1.0  1.0  1.0    1.0    1.0    1.0   1.0   
       Relative Intensity 3 nM      1.0  1.0  1.0    1.0    1.0    1.0   1.0   
       Relative Intensity 30 nM     1.0  1.0  1.0    1.0    1.0    1.0   1.0   

                                                    ...                    \
Gene                              ACOX3 ACP1 ACSL5  ... ULK3 UNC119 VDAC3   
Drug   Concentration                                ...                     
AC-480 Relative Intensity 10 nM     1.0  1.0   1.0  ...  1.0    1.0   1.0   
       Relative Intensity 100 nM    1.0  1.0   1.0  ...  1.0    1.0   1.0   
       Relative Intensity 1000 nM   1.0  1.0   1.0  ...  1.0    1.0   1.0   
       Relative Intensity 3 nM      1.0  1.0   1.0  ...  1.0    1.0   1.0   
       Relative Intensity 30 nM     1.0  1.0   1.0  ...  1.0    1.0   1.0   

                                                                            \
Gene                              WEE1 YARS YES1 YTHDF3;YTHDF1 YWHAG YWHAQ   
Drug   Concentration                                                         
AC-480 Relative Intensity 10 nM    1.0  1.0  1.0           1.0   1.0   1.0   
       Relative Intensity 100 nM   1.0  1.0  1.0           1.0   1.0   1.0   
       Relative Intensity 1000 nM  1.0  1.0  1.0           1.0   1.0   1.0   
       Relative Intensity 3 nM     1.0  1.0  1.0           1.0   1.0   1.0   
       Relative Intensity 30 nM    1.0  1.0  1.0           1.0   1.0   1.0   

                                        
Gene                               ZAK  
Drug   Concentration                    
AC-480 Relative Intensity 10 nM    1.0  
       Relative Intensity 100 nM   1.0  
       Relative Intensity 1000 nM  1.0  
       Relative Intensity 3 nM     1.0  
       Relative Intensity 30 nM    1.0  

[5 rows x 520 columns]

In [60]:
data.iloc[175:200,15:20]

Value                           
Gene                                    ACVRL1     ADCK1 ADCK3 ADCK5 ADD2
Drug        Concentration                                                
AZD-8186    Relative Intensity 10 nM       1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 100 nM      1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 1000 nM     1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 3 nM        1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 30 nM       1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 300 nM      1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 3000 nM     1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 30000 nM    1.0  1.000000   1.0   1.0  1.0
AZD-8330    Relative Intensity 10 nM       1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 100 nM      1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 1000 nM     1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 3 nM        1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 30 nM       1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 300 nM      1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 3000 nM     1.0  1.000000   1.0   1.0  1.0
            Relative Intensity 30000 nM    1.0  1.000000   1.0   1.0  1.0
Abemaciclib Relative Intensity 10 nM       1.0  0.869866   1.0   1.0  1.0
            Relative Intensity 100 nM      1.0  0.000000   1.0   1.0  1.0
            Relative Intensity 1000 nM     1.0  0.000000   1.0   1.0  1.0
            Relative Intensity 3 nM        1.0  1.045974   1.0   1.0  1.0
            Relative Intensity 30 nM       1.0  1.141565   1.0   1.0  1.0
            Relative Intensity 300 nM      1.0  0.000000   1.0   1.0  1.0
            Relative Intensity 3000 nM     1.0  0.000000   1.0   1.0  1.0
            Relative Intensity 30000 nM    1.0  0.000000   1.0   1.0  1.0
Afatinib    Relative Intensity 10 nM       1.0  1.000000   1.0   1.0  1.0

520 columns - need to work on this now. Replaced missing data with '1.0' that will move to zero after a log2 transform. There are zeros in the data as well, so need to add a small pseudocount for these.